cmake/templates/HPXConfig.cmake.in

```diff
@@ -44,8 +42,24 @@ set(HPX_Fortran_COMPILER "@CMAKE_Fortran_COMPILER@" CACHE STRING "Fortran compil
 set(HPX_Fortran_COMPILER_ID "@CMAKE_Fortran_COMPILER_ID@" CACHE STRING "Fortran compiler id for HPX" FORCE)
 set(HPX_Fortran_COMPILER_VERSION "@CMAKE_Fortran_COMPILER_VERSION@" CACHE STRING "Fortran compiler version for HPX" FORCE)
 
+##################################################
+# Setup the imported libraries (publicly linked) #
+set(HPX_PROGRAM_OPTIONS_WITH_BOOST_PROGRAM_OPTIONS_COMPATIBILITY
+  @HPX_PROGRAM_OPTIONS_WITH_BOOST_PROGRAM_OPTIONS_COMPATIBILITY@
+  CACHE STRING "" FORCE)
+set(HPX_FILESYSTEM_WITH_BOOST_FILESYSTEM_COMPATIBILITY
+  @HPX_FILESYSTEM_WITH_BOOST_FILESYSTEM_COMPATIBILITY@
+  CACHE STRING "" FORCE)
+set(HPX_WITH_CXX17_FILESYSTEM @HPX_WITH_CXX17_FILESYSTEM@)
+# Separate targets to be unarily linked to some modules
 set(HPX_BOOST_ROOT "@BOOST_ROOT@")
-set(HPX_BOOST_LIBRARIES "@Boost_LIBRARIES@")
+include(HPX_SetupBoost)
+include(HPX_SetupBoostFilesystem)
+include(HPX_SetupBoostProgramOptions)
+
+set(HPX_HWLOC_ROOT "@HWLOC_ROOT@")
```

Comment:
```suggestion
set(HWLOC_ROOT "@HWLOC_ROOT@")
```

Reply:
Same as above, let's see the results of the tests